source/reference/array.rst

```diff
@@ -236,7 +236,200 @@ Unevaluated Items
+
+.. schema_example::
+
+    {
+            "description": "unevaluatedItems can see annotations from if even without then and else",
+            "schema": {
+                "$schema": "https://json-schema.org/draft/2020-12/schema",
+                "if": {
+                    "prefixItems": [{"const": "a"}]
+                },
+                "unevaluatedItems": false
+            },
+            "tests": [
+                {
+                    "description": "valid in case if is evaluated",
+                    "data": [ "a" ],
+                    "valid": true
+                },
+                {
+                    "description": "invalid in case if is evaluated",
+                    "data": [ "b" ],
+                    "valid": false
+                }
+            ]
+    }
+
+And an important note: ``unevaluatedItems`` can't see inside cousins
+(a vertically adjacent item inside a separate pair of {curly braces}
+with the same "parent"— ``anyOf``, ``if``, ``not``, or similar). Such
+an instance always fails evaluation.
+
+.. schema_example::
+
+    {
+        "description": "unevaluatedItems can't see inside cousins",
+        "schema": {
+            "$schema": "https://json-schema.org/draft/2020-12/schema",
+            "allOf": [
+                {
+                  "prefixItems": [ true ]
+                },
+                { "unevaluatedItems": false }
+            ]
+        },
+        "tests": [
+            {
+                "description": "always fails",
+                "data": [ 1 ],
+                "valid": false
+            }
+        ]
+    }
+
+Finally, here's an example of ``unevaluatedItems`` if you're
+`structuring`. Let's make a "half-closed" schema: something useful
+when you want to keep the first two arguments, but also add more in
+certain situations. ("Closed" to two in some places, "open" to more
+in others.)
+
+.. schema_example::
+
+    {
+      "$id": "https://example.com/my-tuple",
+
+      "type": "array",
+      "prefixItems": [
+        true,
+        { "type": "boolean" }
+      ],
+
+      "$defs": {
+        "closed": {
+          "$anchor": "closed",
+          "$ref": "#",
+          "unevaluatedItems": false
+        }
+      }
+    }
+
+Then we can extend the tuple:
+
+.. schema_example::
+
+    {
+      "$id": "https://example.com/my-extended-tuple",
+
+      "$ref": "/my-tuple",
+      "prefixItems": [
+        true,
+        true,
+        { "type": "boolean" }
+      ],
+      "unevaluatedItems": false
+    }
+
+With this, you can use ``$ref`` to reference the first two
+``prefixItems`` and keep the schema "closed" to two when necessary,
+"open" to more when necessary. A reference to ``/my-tuple#closed``
+would disallow more than two items, when you need it to.
+
+.. note::
+   For a tower of more examples, read our `unevaluatedItems Test Suite <https://github.com/json-schema-org/JSON-Schema-Test-Suite/blob/main/tests/draft2020-12/unevaluatedItems.json>`_ on GitHub.
```

Comment:
"tower" is slightly awkward to me here, I'd say "long list" or if you want to be colorful, "torrent" I guess.

Reply:
Visual words help ideas stick better than abstract words, but sometimes I get too literary. I've been called out on it before by my writing instructor. I agree, "tower" is like a rock in the flow of the sentence.